grouper/fe/handlers/service_account_create.py

```diff
@@ -0,0 +1,112 @@
+from datetime import datetime
+
+from sqlalchemy.exc import IntegrityError
+
+from grouper import group as group_biz
+from grouper.audit import assert_can_join, UserNotAuditor
+from grouper.email_util import send_email
+from grouper.fe.forms import ServiceAccountCreateForm
+from grouper.fe.settings import settings
+from grouper.fe.util import Alert, GrouperHandler
+from grouper.model_soup import (
+        APPROVER_ROLE_INDICIES,
+        Group,
+        GROUP_EDGE_ROLES,
+        GROUP_JOIN_CHOICES,
+        )
+from grouper.models.audit_log import AuditLog
+from grouper.models.user import User
+
+
+class ServiceAccountCreate(GrouperHandler):
+    def get(self):
+        form = ServiceAccountCreateForm()
+        return self.render(
+            "service-account-create.html", form=form,
+        )
+
+    def post(self):
+        form = ServiceAccountCreateForm(self.request.arguments)
+        if not form.validate():
+            return self.render(
+                "service-account-create.html", form=form,
+                alerts=self.get_form_alerts(form.errors)
+            )
+
+        
+        user = User(username=form.data["name"], role_user=True)
+        group = Group(groupname=form.data["name"], description=form.data["description"],
+            canjoin=form.data["canjoin"])
+
+        try:
+            user.add(self.session)
+            self.session.flush()
+        except IntegrityError:
+            self.session.rollback()
+            form.name.errors.append("A user with name {} already exists".format(form.data["name"]))
+            return self.render(
+                "service-account-create.html", form=form,
+                alerts=self.get_form_alerts(form.errors)
+            )
+
+        try:
+            group.add(self.session)
+            self.session.flush()
+        except IntegrityError:
+            self.session.rollback()
+            form.name.errors.append("A group with name {} already exists".format(form.data["name"]))
+            return self.render(
+                "service-account-create.html", form=form,
+                alerts=self.get_form_alerts(form.errors)
+            )
+
+        group.add_member(self.current_user, self.current_user, "GroupCreator",
+            "actioned", None, "np-owner")
+        group.add_member(self.current_user, user, "Service Account",
+            "actioned", None, "member")
+        self.session.commit()
+
+        AuditLog.log(self.session, self.current_user.id, 'create_group',
+                     'Created new service account.', on_group_id=group.id)
+
+        return self.redirect("/groups/{}?refresh=yes".format(group.name))
+
+    def _get_member(self, member_choice):
```

Comment:
They don't do anything; they were mistakenly left in and then deleted in a later commit.